### DISCUSSION TITLE
Atomic composability between Subnets and C-Chain

### DISCUSSION THREAD

**Discussion by hyperserac** [ava-labs/hypersdk#1284] [Chat]
Transactional atomicity (execution + state commitment) between Subnets and the C-Chain could bring a lot of usecases. 
Low latency WARP messages are already enhancing the UX quite a lot but atomic composability brings also good devX.
I know it's a very broad topic but is it part of a roadmap ?

**Comment by aaronbuchwald** [comment]
This is not currently part of the roadmap. L1s on Avalanche are inherently independent, so interchain messaging (ICM) between them is async.

To support atomic composability, that would require either application level locking (probably not a great idea) or a shared sequencer for a cluster of applications where there's a demand for atomic composability over async composability offered by ICM.

Related tweet from @mtgnoah https://x.com/noah_pravecek/status/1821981366495670552

cc @michaelkaplan13
